integration/token/fungible/tests.go

```diff
@@ -716,6 +719,51 @@ func TransferCash(network *integration.Infrastructure, id string, wallet string,
 		time.Sleep(5 * time.Second)
 	}
 }
+func TransferCashAndCheckRecepientSignature(network *integration.Infrastructure, id string, wallet string, typ string, amount uint64, receiver string, auditor string, errorMsgs ...string) {
+	fmt.Println("TransferCashAndCheckRecepientSignature")
+
+	txid, err := network.Client(id).CallView("transfer", common.JSONMarshall(&views.Transfer{
+		Auditor:   auditor,
+		Wallet:    wallet,
+		Type:      typ,
+		Amount:    amount,
+		Recipient: network.Identity(receiver),
+	}))
+
+	if len(errorMsgs) == 0 {
+		Expect(err).NotTo(HaveOccurred())
+		Expect(network.Client(receiver).IsTxFinal(common.JSONUnmarshalString(txid))).NotTo(HaveOccurred())
+		Expect(network.Client("auditor").IsTxFinal(common.JSONUnmarshalString(txid))).NotTo(HaveOccurred())
+
+	} else {
+		Expect(err).To(HaveOccurred())
+		for _, msg := range errorMsgs {
+			Expect(err.Error()).To(ContainSubstring(msg))
+		}
+		time.Sleep(5 * time.Second)
+	}
+	CheckRecipientSignature(network, id, common.JSONUnmarshalString(txid), receiver)
+}
+func CheckRecipientSignature(network *integration.Infrastructure, id string, txnId string, receiver string, errorMsgs ...string) {
```

Comment:
shall we add this check directly to TransferCash instead of introducing TransferCashAndCheckRecepientSignature?